core/src/test/scala/org/broadinstitute/dsde/rawls/mock/MockSamDAO.scala

```diff
@@ -13,8 +13,10 @@ class MockSamDAO(dataSource: SlickDataSource)(implicit executionContext: Executi
 
   override def registerUser(userInfo: UserInfo): Future[Option[RawlsUser]] = ???
 
-  override def getUserStatus(userInfo: UserInfo): Future[Option[RawlsUser]] =
-    Future.successful(Option(RawlsUser(userInfo)))
+  override def getUserStatus(userInfo: UserInfo): Future[Option[SamUserStatusResponse]] =
```

Comment:
Re-using `RawlsUser` as the return type was forcing me to tack on an `enabled` val to that case class, with a default of `true` (see @marctalbott 's point earlier in the PR). Since there are only two call sites for this method, shifting to a first-class SAM model seems semantically clearer.